docs/develop/dotnet/message-passing.mdx

```diff
@@ -1,261 +1,791 @@
+Use dynamic elements judiciously and as a fallback mechanism, not a primary design.
+They can introduce long-term maintainability and debugging issues.
+Reserve dynamic invocation use for cases where a name is not or can't be known at compile time.
+
+:::
+
+### Set a dynamic Signal, Query, or Update handler {#set-a-dynamic-signal}
+
+A dynamic Signal, Query, or Update refers to a special stand-in handler.
+It's used when an unregistered handler request arrives.
+
+<div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**Needs .NET.**</div>
+Consider a Signal, where you might send something like  `workflow.signal(MyWorkflow.my_signal_method, my_arg)`.
+This is a type-safe compiler-checked approach that guarantees a method exists.
+There's also a non-type-safe string-based form: `workflow.signal('some-name', my_arg)`.
+When sent to the server, the name is checked only after arriving at the Worker.
+This is where "dynamic handlers" come in.
+
+After failing to find a handler with a matching name and type, the Worker checks for a registered dynamic stand-in handler.
+If found, the Worker uses that instead.
+
+You must opt handlers into dynamic access.
+For example, you make a Signal dynamic by adding `Dynamic = true` to the `[WorkflowSignal]` attribute.
+
+The handler must accept a string input for the name, and a `Temporalio.Converters.IRawValue[]` for the arguments.
+Use a [Workflow.PayloadConverter](https://dotnet.temporal.io/api/Temporalio.Workflows.Workflow.html#Temporalio_Workflows_Workflow_PayloadConverter)  to convert `IRawValue` objects to your required type.
+For example:
 
 ```csharp
-[WorkflowUpdate]
-public async Task<string> UpdateStatusAsync(Status status)
+[WorkflowSignal(Dynamic = true)]
+public async Task DynamicSignalAsync(string signalName, IRawValue[] args)
 {
-    this.status = status;
-    return "Status updated";
+    var input = Workflow.PayloadConverter.ToValue<DoSomethingParam>(args.Single());
+    pendingThings.Add(input);
 }
 ```
 
-### Send an Update from a Temporal Client {#send-update-from-client}
+This sample creates a dynamic Signal.
+When an unregistered or unrecognized Signal arrives with a matching signature, dynamic assignment uses this handler to manage the Signal.
+It is responsible for transforming the sequence contents into usable data in a form that the method's logic can process and act on.
+
+When registered, any Signals sent to the Workflow without a defined handler will be delivered to the dynamic `WorkflowSignal`.
+Note that you can only register one each of a dynamic Signal, Query, or Update per Workflow Execution.
+Dynamic handlers can be implemented in both regular and dynamic Workflow implementations.
+
+
+<div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**.NET source did not include dynamic Workflow coverage.**</div>
```

Comment:
Dynamic workflows and activities are with workflow and activity docs instead of with query, signal, update docs